lib/src/scanner_manager.dart

```diff
@@ -0,0 +1,84 @@
+import 'dart:async';
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/services.dart';
+
+/// Manager of information read by the scanner and sent as keyboard events.
+class ScannerManager {
+  /// Due to information coming as events, we wait for the next event for some time.
+  /// This value is the default duration between events in the data sequence read once.
+  static const Duration _defaultDuration = Duration(milliseconds: 300);
+
+  /// The event source for devices that work as a hardware keyboard.
+  ///
+  /// See https://developer.android.com/reference/android/view/InputDevice#SOURCE_KEYBOARD
+  static const int _hardwareKeyboardSourceEvent = 257;
+
+  /// List of default target event sources, if special doesn't set
```

Comment:
what do you mean by "if special doesn't set"?

Reply:
If the user passes his own list to the constructor